File: bots/messenger-bot.ts
import { BotManager } from "./bot-manager";

export abstract class MessengerBot {
  protected botManager: BotManager;

  // sends the message to all chats that subscribed at least one of the specified channels
  // returns the number of chats the message was sent to
  abstract sendMessageToChannels(message: string, channels: string[]): number;

  // method to register a bot manager (after instantiation)
  registerBotManager(botManager: BotManager): void {
    this.botManager = botManager;
  }

  // TODO: get chats?, get chats per channel? (-> how many chats are subscribed to each channel on which messenger), ...
}

File: bots/messenger_bot.py
from typing import List


# TOOD: resolve cyclic dependency

class MessengerBot:

	botManager = None

	# sends the message to all chats that subscribed at least one of the specified channels
	# returns the number of chats the message was sent to
	def sendMessageToChannels(self, message: str, channels: List[str]) -> int:
		raise NotImplementedError("Abstract Method")

	# method to register a bot manager (after instantiation)
	def registerBotManager(self, botManager) -> None:
		self.botManager = botManager
